src/zarr/core/array.py

```diff
@@ -3448,4 +3500,367 @@ def _get_default_codecs(
     else:
         dtype_key = "numeric"
 
-    return [{"name": codec_id, "configuration": {}} for codec_id in default_codecs[dtype_key]]
+    return cast(list[dict[str, JSON]], default_codecs[dtype_key])
+
+
+FiltersParam: TypeAlias = (
+    Iterable[dict[str, JSON] | ArrayArrayCodec]
+    | ArrayArrayCodec
+    | Iterable[numcodecs.abc.Codec]
+    | Literal["auto"]
+)
+CompressorsParam: TypeAlias = (
+    Iterable[dict[str, JSON] | BytesBytesCodec]
+    | BytesBytesCodec
+    | numcodecs.abc.Codec
+    | Literal["auto"]
+)
+
+
+async def create_array(
+    store: str | StoreLike,
+    *,
+    name: str | None = None,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunks: ChunkCoords | Literal["auto"] = "auto",
+    shards: ChunkCoords | Literal["auto"] | None = None,
+    filters: FiltersParam = "auto",
+    compressors: CompressorsParam = "auto",
```

Comment:
How can a user specify that they don't want any filters or compressors? Overriding the defaults. Would that be `None` or `()`? I think we should allow both.

Reply:
See https://github.com/zarr-developers/zarr-python/pull/2463/commits/c4330ef3778270e7384d0e14e5402af21426f203#diff-d1499eae500b1036f9aa3e3c6266bc516b9f03efcd65644acac5418e406822baR968-R1003